full-stack-tests/presentation/src/learning-snippets/Hierarchies.test.ts

```diff
@@ -24,305 +28,1613 @@ describe("Learning Snippets", () => {
+          const ruleset: Ruleset = {
+            id: "example",
+            rules: [{
+              ruleType: RuleTypes.RootNodes,
+              specifications: [{
+                specType: ChildNodeSpecificationTypes.CustomNode,
+                type: "T_MY_NODE",
+                label: "My Node",
+              }],
+            }],
+          };
+          // __PUBLISH_EXTRACT_END__
+          printRuleset(ruleset);
+
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.Label.Result
+          // Verify that node with correct label is returned
+          const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+          expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+            label: { displayValue: "My Node" },
+          }]);
+          // __PUBLISH_EXTRACT_END__
+        });
+
+        it("uses `description` attribute", async () => {
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.Description.Ruleset
+          // The ruleset has a root node specification that returns a single custom node and assigns it a description.
+          const ruleset: Ruleset = {
+            id: "example",
+            rules: [{
+              ruleType: RuleTypes.RootNodes,
+              specifications: [{
+                specType: ChildNodeSpecificationTypes.CustomNode,
+                type: "T_MY_NODE",
+                label: "My Node",
+                description: "My node's description",
+              }],
+            }],
+          };
+          // __PUBLISH_EXTRACT_END__
+          printRuleset(ruleset);
+
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.Description.Result
+          // Verify that node with correct description is returned
+          const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+          expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+            description: "My node's description",
+          }]);
+          // __PUBLISH_EXTRACT_END__
+        });
+
+        it("uses `imageId` attribute", async () => {
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.ImageId.Ruleset
+          // The ruleset has a root node specification that returns a single custom node and assigns it an image identifier.
+          const ruleset: Ruleset = {
+            id: "example",
+            rules: [{
+              ruleType: RuleTypes.RootNodes,
+              specifications: [{
+                specType: ChildNodeSpecificationTypes.CustomNode,
+                type: "T_MY_NODE",
+                label: "My Node",
+                imageId: "my-icon-identifier",
+              }],
+            }],
+          };
+          // __PUBLISH_EXTRACT_END__
+          printRuleset(ruleset);
+
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.ImageId.Result
+          // Verify that node with correct image identifier is returned
```

Comment:
Nitpick: Missing  "." at the end. Similar case with other *..Result examples

Reply:
This is a code example. I don't think there's a requirement for code comments to have a `.` at the end of a sentence. I think that's only required when there's more than 1 sentence.